src/me/drton/jmavsim/SimpleSensors.java

```diff
@@ -37,13 +37,18 @@ public Vector3d addZeroMeanNoise(Vector3d vIn, double var) {
 
         x0 = java.lang.Math.sqrt(-2.0 * java.lang.Math.log(b0)) * java.lang.Math.cos(Math.PI * 2.0 * b1);
 
-        double n = x0 * (var * var);
-
-        if (java.lang.Double.isFinite(n)) {
-            return new Vector3d(vIn.x + n, vIn.y + n, vIn.z + n);
-        } else {
-            return vIn;
+        if (java.lang.Double.isInfinite(x0) || java.lang.Double.isNaN(x0)) {
+            x0 = 0.0;
         }
+
+        return x0 * (stdDev * stdDev);
```

Comment:
I only had a very fast read through. Are you sure it's squared here? https://en.wikipedia.org/wiki/Box%E2%80%93Muller_transform#Implementation

Can we not use a native java function like: https://docs.oracle.com/javase/7/docs/api/java/util/Random.html#nextGaussian()